### PR TITLE
x972- SPRI Clean up feature

### PR DESCRIPTION
**Pull request by seenanair**
SPRI Clean up feature
- Cleanup component
- VisiumQC page modification

### CODE REVIEW

File: src/components/visiumQC/Cleanup.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          <div className="flex flex-row items-center justify-end">
+            {<RemoveButton data-testid={'remove'} type="button" onClick={() => removeLabware(labware.barcode)} />}
+          </div>
+          <div className="flex flex-col items-center justify-around">
+            {/* Display the layout of the labware */}
+            <div className="bg-blue-100" data-testid={'labware'}>
+              <Labware labware={labware} slotBuilder={slotBuilder} />
+            </div>
+
+            {/* Display the list of pass/fail comments */}
```
Comment (khelwood) [inline src/components/visiumQC/Cleanup.tsx:101]: What does pass/fail refer to here?

File: src/pages/Configuration.tsx [hunk 2]
```diff
@@ -19,6 +20,7 @@ export default function Configuration({ configuration }: ConfigurationParams) {
   const groupedComments = groupBy(configuration.comments, 'category');
   const groupedEquipments = groupBy(configuration.equipments, 'category');
 
+  debugger;
```
Comment (khelwood) [inline src/pages/Configuration.tsx:23]: Presumably remove `debugger;`

File: src/components/visiumQC/Cleanup.tsx (hunk 1 above)
Comment (seenanair) [reply src/components/visiumQC/Cleanup.tsx:101]: That's a comment which is not updated, it is changed now to 'Display the dropdown to set/reset all comments'

File: src/pages/VisiumQC.tsx [hunk 3]
```diff
@@ -157,17 +199,26 @@ export default function VisiumQC({ info }: VisiumQCProps) {
         const val = value.slotMeasurements.filter((measurement) => measurement.value === '');
         return val.length <= 0;
       } else return false;
-    } else {
+    }
+    if (value.qcType === QCType.SLIDE_PROCESSING) {
       return !!value.labwareResult;
     }
+    if (value.qcType === QCType.SPRI_CLEANUP) {
+      return value.slotComments && value.slotComments.map((sc) => sc.commentId >= 0).length > 0;
```
[the remark this answers is not in the document]
Comment (seenanair) [reply src/pages/VisiumQC.tsx:207]: This is to check whether there is at-least one comment selected by the user for a slot and this is used to set the enable/disable status of Save button and if not 'Save' button will be disabled.

[the remark this answers is not in the document]
Comment (khelwood) [reply src/pages/VisiumQC.tsx:207]: I don't think `map` works that way. I believe the result of `map` will have the same length as the original array. Perhaps `filter` would work.

[the remark this answers is not in the document]
Comment (seenanair) [reply src/pages/VisiumQC.tsx:207]: Sorry, it is really stupid of me. I don't know what I was thinking. You are right

[the remark this answers is not in the document]
Comment (khelwood) [reply src/pages/VisiumQC.tsx]: Anyone can make mistakes

**Review by khelwood — APPROVED** [review]

File: src/pages/VisiumQC.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (seenanair) [reply src/pages/VisiumQC.tsx]: Thank you. Strangely, I didn't even notice it while explaining to you for the comment

[the remark this answers is not in the document]
Comment (khelwood) [reply src/pages/VisiumQC.tsx]: I think the preferable way to write this is probably with [Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some), e.g. `value.slotComments.some(sc =>  sc.commentId >= 0);`

[the remark this answers is not in the document]
Comment (seenanair) [reply src/pages/VisiumQC.tsx]: Updated

**Review by khelwood — APPROVED** [review]

**Pull request merged**